repository-activity-score.md

```diff
@@ -0,0 +1,122 @@
+    // boost calculation:
+    // all repositories updated in the previous year will receive a boost of maximum 1000 declining by days since last update
+    let iBoost = (1000 - Math.min(iDaysSinceLastUpdate, 365) * 2.74);
+    // gradually scale down boost according to repository creation date to mix with "real" engagement stats
+    let iDaysSinceCreation = (new Date().getTime() - new Date(repo.created_at).getTime()) / 1000 / 86400;
+    iBoost *= (365 - Math.min(iDaysSinceCreation, 365)) / 365;
+    // add boost to score
+    iScore += iBoost;
+    // give projects with contribution guidelines (CONTRIBUTING.md) file a static boost of 100
+    iScore += (repo["_InnerSourceMetadata"] && repo["_InnerSourceMetadata"]["guidelines"] ? 100 : 0);
+    // build in a logarithmic scale for very active projects (open ended but stabilizing around 5000)
+    if (iScore > 3000) {
+        iScore = 3000 + Math.log(iScore) * 100;
+    }
+    // final score is a rounded value starting from 0
+    iScore = Math.round(iScore - 1);
+    // add score to metadata on the fly
+    repo._InnerSourceMetadata = repo._InnerSourceMetadata || {};
+    repo._InnerSourceMetadata.score = iScore;
+    return iScore;
+}
+```
+
+## Resulting Context
+
+
+Contributors are free to commit a part of their time to InnerSource project. They may choose to contribute to a project that they depend on for the work in their regular team anyways. However they may also choose to contribute to something completely different, based on their interests and personal development goals.
+
+Projects can be sorted and presented by repository activity score to give a meaningful order in a portal presenting projects to potential new contributors. The score can be calculated on the fly or in a background job that evaluates all projects on a regular basis and stores a list of results.
+
+A crawler that regularly searches all InnerSource repositories (e.g. tagged with a certain topic in GitHub) can be a helpful addition as well. It provides a ranked list of projects that can be used as an input for tools like the [InnerSource Portal](innersource-portal.md), a search engine, or an interactive chat bot. 
+
+## Rationale
+
+The repository activity score is a simple calculation based on the GitHub API. It can be fully automated and easily adapted to new requirements. 
+
+## Known Instances
+
+Used in SAP's InnerSource project portal to define the default order of the InnerSource projects. It was first created in July 2020 and is fine-tuned and updated frequently ever since.
+
+When proposed to InnerSourceCommons in July 2020, this pattern emerged.
+
+## Status
+
+
+* First Draft: 30th July 2020  
+* Second Draft: 5th August 2020  
+* Third Draft: 6th August 2020  
+
+## Author(s)
+
+[Michael Graf (SAP)](mi.graf@sap.com)
+
+## Acknowledgements
+
+Thank you to the InnerSource Commons Community for lightning-fast advice, and a lot of helpful input to feed this pattern!
```

Comment:
Nice :D 
I added a shout out to the Slack to find people that feel like they want to be named as contributors.
What do the existing reviewers of this PR think? What does @Michadelic think about adding named acknowledgements? (They're present in other patterns, hence my question.)

Reply:
great idea, will add acknowledgements right away, thanks again!